src/staking/RewardTracker.sol

```diff
@@ -0,0 +1,237 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.10;
+
+import {ERC20} from "solmate/tokens/ERC20.sol";
+import {SafeTransferLib} from "solmate/utils/SafeTransferLib.sol";
+import {FixedPointMathLib} from "solmate/utils/FixedPointMathLib.sol";
+import {AccessControl} from "openzeppelin-contracts/access/AccessControl.sol";
+import {ReentrancyGuard} from "openzeppelin-contracts/security/ReentrancyGuard.sol";
+import {BonusTracker} from "./BonusTracker.sol";
+
+contract RewardTracker is BonusTracker, AccessControl {
+    using SafeTransferLib for ERC20;
+    using FixedPointMathLib for uint256;
+
+    event RewardAdded(uint256 reward);
+    event RewardPaid(address indexed user, uint256 reward);
+
+    error Error_AmountTooLarge();
+
+    /// @notice The last Unix timestamp (in seconds) when rewardPerTokenStored was updated
+    uint64 public lastUpdateTime;
+
+    /// @notice The Unix timestamp (in seconds) at which the current reward period ends
+    uint64 public periodFinish;
+
+    /// @notice The per-second rate at which rewardPerToken increases
+    uint256 public rewardRate;
+
+    /// @notice The last stored rewardPerToken value
+    uint256 public rewardPerTokenStored;
+
+    /// @notice Role allowed to call notifyReward()
+    bytes32 public constant DISTRIBUTOR = keccak256("DISTRIBUTOR");
+
+    /// @notice The rewardPerToken value when an account last staked/withdrew/withdrew rewards
+    mapping(address => uint256) public userRewardPerTokenPaid;
+
+    /// @notice The earned() value when an account last staked/withdrew/withdrew rewards
+    mapping(address => uint256) public rewards;
+
+    /// @notice The token being rewarded to stakers
+    ERC20 public immutable rewardToken;
+
+    /// @notice The length of each reward period, in seconds
+    uint64 immutable DURATION;
+
+    constructor(address _stakeToken, string memory _name, string memory _symbol, address _rewardToken, uint64 _DURATION)
+        BonusTracker(ERC20(_stakeToken), _name, _symbol)
+    {
+        _grantRole(DEFAULT_ADMIN_ROLE, msg.sender);
+        rewardToken = ERC20(_rewardToken);
+        DURATION = _DURATION;
+    }
+
+    function totalAssets() public view override returns (uint256 assets) {
+        assets = totalSupply;
+    }
+
+    function deposit(uint256 assets, address receiver) public override returns (uint256 shares) {
+        _updateReward(receiver);
+        _updateBonus(receiver);
+        shares = super.deposit(assets, receiver);
+    }
+
+    function mint(uint256 assets, address receiver) public override returns (uint256 shares) {
```

Comment:
Why does the `mint` function have the same implementation as the `deposit` function?

Reply:
fixed :roller_skate: